### PR TITLE
TE-3.1_fix : gribi programming support with mac and ipaddress combination

### PR DESCRIPTION
**Pull request by manan-patel**
Adding gribi NH programming support for mac and ip address combination, which fixes TE-3.1 TC

Dummy address is needed for gribi NH programming with mac address, with dummy address present using interfaceref or subinterfaceref with mac for NH programming will lead to incomplete forwarding and traffic will end up dropping.

Having nh dummy address plus mac fixes this problem.

### CODE REVIEW

File: internal/gribi/gribi.go [hunk 1]
```diff
@@ -225,11 +225,15 @@ func (c *Client) AddNH(t testing.TB, nhIndex uint64, address, instance string, e
 		}
 	case "MACwithInterface":
 		for _, opt := range opts {
-			if opt.SubInterface != 0 {
-				nh = nh.WithSubinterfaceRef(opt.Interface, opt.SubInterface).
-					WithMacAddress(opt.Mac)
-			} else {
+			if opt.Dest != "" {
 				nh = nh.WithIPAddress(opt.Dest).WithMacAddress(opt.Mac)
```
Comment (LimeHat) [inline internal/gribi/gribi.go:229]: This statement will not use interface-ref at all though.
Is this the intention?

If you want just to supplement interface-ref with an ip. addr, I think Ankur resolves this in a more appropriate way in #1762

Comment (manan-patel) [reply internal/gribi/gribi.go:229]: Correct, I was checking if we have Destination ip provided in which case we use mac + ip
else we further check if subinterface is there and use subinterface + Mac else Interface+Mac

Comment (LimeHat) [reply internal/gribi/gribi.go:229]: In that case (if interface-ref cannot be supported / Ankur's PR is not going to solve it), I'd propose to do this as a separate `case` statement in the helper.

Overloading `MACwithInterface` case with a scenario where, in fact, interface-reference is not used or present is very confusing/misleading.

Comment (manan-patel) [reply internal/gribi/gribi.go:229]: Okay, let me add another case

Comment (LimeHat) [reply internal/gribi/gribi.go]: thanks 👍

**Comment by guoshiuan** [comment]
b/288899973 has the internal test results.

**Review by guoshiuan — APPROVED** [review]

**Pull request merged**